agent/worker.go

```diff
@@ -500,8 +505,16 @@ func (w *worker) newTaskManager(ctx context.Context, tx *bolt.Tx, task *api.Task
 // updateTaskStatus reports statuses to listeners, read lock must be held.
 func (w *worker) updateTaskStatus(ctx context.Context, tx *bolt.Tx, taskID string, status *api.TaskStatus) error {
 	if err := PutTaskStatus(tx, taskID, status); err != nil {
+		// we shouldn't fail to put a task status. however, there exists the
+		// possibility of a race in which we try to put a task status after the
+		// task has been deleted. because this whole contraption is a careful
+		// dance of too-tightly-coupled concurrent parts, fixing tht race is
+		// fraught with hazards. instead, we'll recognize that it can occur,
+		// log the error, and then ignore it.
 		log.G(ctx).WithError(err).Error("failed writing status to disk")
-		return err
+		if err != errTaskUnknown {
+			return err
```

Comment:
Not familiar with all this, but should we return `nil` in case it's `errTaskUnknown`? Currently it looks like we'll continue processing below; is that correct if the task was deleted?